### PR TITLE
[PS-1783] Fix file selector input bug from PS-1465 ( #3502 )

### PR DESCRIPTION
**Pull request by schang1146**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Fixes the the file selector input showing up twice because of CSP violations.

## Code changes

- **apps/web/src/app/tools/import-export/import.component.html:**
  - Use HTML5 [hidden] attribute
  - Use Angular (click) event binding and element variable
  - Added [disabled] attribute that used to be on the other file selector input element

## Screenshots

https://user-images.githubusercontent.com/20074034/198714722-3a05326b-0cda-457b-850b-7214168b7bbd.mp4

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Comment by schang1146** [comment]
@djsmith85 
I noticed another possible issue with large filenames:
image

Currently, in the live web app, the name is shortened automatically:
image

Is this something you'd want to be fixed before pushing the new button out?

**Review by djsmith85 — CHANGES_REQUESTED** [review]
@schang1146 Thank you so much for coming back on this.

Interesting, the filename does get shortened on the currently deployed version. Maybe something the browser does with file-inputs, similar to the translation of the button when the browsers UI language is set to another language.

With your changes, my filename get wrapped instead of showing on a new line like your's does 🤷 
image

I personally don't think this is a big issue, as you'll likely won't be importing often and in any case a single file at a  time.

The merge conflict though, that is likely caused by me reverting your changes and merging into master. Please pull master again and fix the conflict. Then I'll have another look at it.

**Comment by schang1146** [comment]
Yep, having <input type="file"> it'll handle shortening the selected filename itself. Not sure how the element decides this but it looks like it's always able to tell how much width is available and shorten it by inserting an ellipsis in the middle of the filename. Super cool stuff but I haven't been able to find how it's implemented.

@djsmith85 - merged master into the branch and fixed the conflicts :)

**Review by djsmith85 — CHANGES_REQUESTED** [review]
I think something went wrong with fixing the merge-conflict. I had re-verted your previous changes and with this you would need to include the changes on the apps/web/src/app/tools/import-export/import.component.ts again.

For example the `fileSelected`-property is missing, which you added here:

https://github.com/schang1146/bitwarden-clients/blob/1b594d25b8ef70958ee7725161f714116a4aafb8/apps/web/src/app/tools/import-export/import.component.ts#L28

**Comment by schang1146** [comment]
Added the changes to `import.component.ts` back.

Thanks for catching that! I missed the merge from master would not throw any conflicts for that file and revert it.

**Review by djsmith85 — DISMISSED** [review]
@schang1146 Changes are looking good now, and after testing it myself, I am passing this on to QA for approval.

Thank you for following up on this.

**Comment by schang1146** [comment]
Saw that there was a small merge conflict, hope you don't mind me pushing a quick fix for it. Lmk if I should wait next time!

File: apps/web/src/app/tools/import-export/import.component.html [hunk 1]
```diff
@@ -291,11 +291,26 @@ <h1>{{ "importData" | i18n }}</h1>
     <div class="col-6">
       <div class="form-group">
         <label for="file">2. {{ "selectImportFile" | i18n }}</label>
+        <br />
+        <div class="file-selector">
+          <button
+            type="button"
+            class="btn btn-outline-primary"
+            (click)="fileSelector.click()"
+            [disabled]="importBlockedByPolicy$ | async"
```
Comment (djsmith85) [inline apps/web/src/app/tools/import-export/import.component.html:300]: @schang1146  The merge commit looks good but for some reason you accepted yours over the incoming changes.

The use of the importBlockedByPolicy$ observable and async pipe got removed with https://github.com/bitwarden/clients/pull/4010

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: apps/web/src/app/tools/import-export/import.component.html (hunk 1 above)
Comment (schang1146) [reply apps/web/src/app/tools/import-export/import.component.html]: Ope, my bad. Missed a stray call. I tried accepting incoming changes but it didn't conflict with that one.

**Review by djsmith85 — APPROVED** [review]
Thank you for the fix. Approving this again and waiting for the okay from QA.

**Comment by djsmith85** [comment]
@schang1146 Just received approval from QA. This is looking good and will be included in the 2022.12 release. Great work on this 👍

**Pull request merged**